Brokerages/Alpaca/PricingStreamSession.cs

```diff
@@ -0,0 +1,96 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace QuantConnect.Brokerages.Alpaca
+{
+	internal class PricingStreamSession
+	{
+		private AlpacaApiBase _api;
+		private List<string> _instruments;
+
+		private bool _shutdown;
+		private Task _runningTask;
+		private Markets.NatsClient _client;
+
+		/// <summary>
+		/// The event fired when a new message is received
+		/// </summary>
+		public event Action<Markets.IStreamQuote> QuoteReceived;
+
+		public PricingStreamSession(AlpacaApiBase alpacaApiBase, List<string> instruments)
+		{
+			this._api = alpacaApiBase;
+			this._instruments = instruments;
+		}
+
+
+		public void StartSession()
+		{
+			_shutdown = false;
+
+			_client = _api.GetNatsClient();
+
+			_runningTask = Task.Run(() =>
+			{
+				_client.Open();
+                _client.QuoteReceived += QuoteReceived;
```

Comment:
I think we also need to attach an event handler for the `TradeReceived` event here, so we can receive both trades and quotes.